src/astro/databases/base.py

```diff
@@ -0,0 +1,259 @@
+            self.get_table_qualified_name(target_table), statement
+        )
+        self.hook.run(statement, parameters)
+
+    def drop_table(self, table: Table) -> None:
+        """
+        Delete a SQL table, if it exists.
+
+        :param table: The table to be deleted.
+        """
+        statement = self._drop_table_statement.format(
+            self.get_table_qualified_name(table)
+        )
+        self.hook.run(statement)
+
+    # ---------------------------------------------------------
+    # Table load methods
+    # ---------------------------------------------------------
+    def load_file_to_table(
+        self,
+        source_file: str,
+        target_table: Table,
+        if_exists: LoadExistStrategy = "replace",
+        chunk_size: int = DEFAULT_CHUNK_SIZE,
+    ) -> None:
+        """
+        Create a table with the file's contents.
+        If the table already exists, append or replace the content, depending on the value of `if_exists`.
+
+        :param source_file: Local or remote filepath
+        :param target_table: Table in which the file will be loaded
+        :param if_exists: Strategy to be used in case the target table already exists.
+        :param chunk_size: Specify the number of rows in each batch to be written at a time.
+        """
+        raise NotImplementedError
+
+    def load_pandas_dataframe_to_table(
+        self,
+        source_dataframe: pd.DataFrame,
+        target_table: Table,
+        if_exists: LoadExistStrategy = "replace",
+        chunk_size: int = DEFAULT_CHUNK_SIZE,
+    ) -> None:
+        """
+        Create a table with the dataframe's contents.
+        If the table already exists, append or replace the content, depending on the value of `if_exists`.
+
+        :param source_dataframe: Local or remote filepath
+        :param target_table: Table in which the file will be loaded
+        :param if_exists: Strategy to be used in case the target table already exists.
+        :param chunk_size: Specify the number of rows in each batch to be written at a time.
+        """
+        source_dataframe.to_sql(
+            target_table.name,
+            con=self.sqlalchemy_engine,
+            if_exists=if_exists,
+            chunksize=chunk_size,
+            method="multi",
+            index=False,
+        )
+
+    def append_table(
+        self,
+        source_table: Table,
+        target_table: Table,
+        if_conflicts: AppendConflictStrategy = "exception",
+    ):
+        """
+        Append the source table rows into a destination table.
+        The argument `if_conflicts` allows the user to define how to handle conflicts.
+
+        :param source_table: Contains the rows to be appended to the target_table
+        :param target_table: Contains the destination table in which the rows will be appended
+        :param if_conflicts: The strategy to be applied if there are conflicts.
+        """
+        raise NotImplementedError
```

Comment:
Why is the append function a part of the database class?

Reply:
My understanding was that any operation directly affecting databases - which may be customized depending on the database, should be implemented within the Database class. Do you see any benefit in extracting this feature outside of the Database class, @dimberman ?